### PR TITLE
Switch to uv

### PR DESCRIPTION
**Pull request by blackary**
If you want 😃

### CODE REVIEW

**Comment by arnaudmiribel** [comment]
Oh absolutely! Can you please revive the `requirements.txt` though? Cloud doesn't yet support uv  https://github.com/streamlit/streamlit/issues/9502

File: hello.py
Comment (arnaudmiribel) [inline hello.py:1]: Sorry, why is this needed?

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[project]
+name = "streamlit-dataframe-summary"
+version = "0.1.0"
+description = "Streamlit component to summarize a DataFrame"
+readme = "README.md"
+requires-python = ">=3.12"
```
Comment (arnaudmiribel) [inline pyproject.toml]: I think we could have less strict python requirement

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+authors = [{ name = "Arnaud Miribel", email = "arnaudmiribel@gmail.com" }]
+dependencies = ["plotly>=5.24.1", "seaborn>=0.13.2", "streamlit>=1.38"]
```
Comment (arnaudmiribel) [inline pyproject.toml:8]: I'm fine with having `streamlit>=1.38` for now, especially since I'm using un-experimental commands for st.dialog and st.fragment - but that could easily be supported for earlier versions though.

Comment (arnaudmiribel) [reply pyproject.toml:8]: For what it's worth, I have uploaded a separate version for our SiS apps that uses experimental commands, and also doesn't use Material icons

File: requirements.txt
Comment (arnaudmiribel) [inline requirements.txt:1]: Can you revive this file? I think it is still _required_ (hehe) for now so that Streamlit Cloud picks up dependencies properly https://github.com/streamlit/streamlit/issues/9502

**Review by arnaudmiribel — CHANGES_REQUESTED** [review]
Thanks @blackary. Tiny request for the `requirements.txt`, otherwise LGTM - wanna deploy a branch app just to validate?

File: hello.py
Comment (blackary) [reply hello.py:1]: Whoops, that was auto-generated by `uv init` :) I'll remove it

File: pyproject.toml (hunk 1 above)
Comment (blackary) [reply pyproject.toml]: Yup, I'll soften that

**Comment by blackary** [comment]
@arnaudmiribel I've addressed the comments, I believe

**Review by arnaudmiribel — APPROVED** [review]
Yay!
